client/core/trade.go

```diff
@@ -2043,6 +2030,11 @@ func (c *Core) redeemMatchGroup(t *trackedTrade, matches []*matchTracker, errs *
 			match.Status = order.MakerRedeemed
 			proof.MakerRedeem = coinID
 		}
+		if t.isMarketBuy() {
+			t.unlockRedemptionFraction(1, uint64(len(t.matches)))
+		} else {
+			t.unlockRedemptionFraction(match.Quantity, t.Trade().Quantity)
+		}
```

Comment:
How about just sum the `match.Quantity` while you loop so you can call your unlock functions just once at the end?